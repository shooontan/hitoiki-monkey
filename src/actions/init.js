/**
 * timeline
 *  - type: 'speech' | 'error'
 *  - text: string
 *  - isFinal?: boolean
 *  - ts?: number
 *
 */
export const Init = () => ({
  calc: false,
  enable: !!SpeechRecognition,
  status: '',
  error: null,
  started: false,
  timeline: [],
});
