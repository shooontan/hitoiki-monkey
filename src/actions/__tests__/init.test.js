import { Init } from '../init';

global.SpeechRecognition = true;

test('Init action', () => {
  const state = Init();
  expect(state).toMatchSnapshot();
});
