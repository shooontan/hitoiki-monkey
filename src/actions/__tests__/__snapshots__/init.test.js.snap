// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init action 1`] = `
Object {
  "calc": false,
  "enable": true,
  "error": null,
  "started": false,
  "status": "",
  "timeline": Array [],
}
`;
